src/blk/aio/aio.cc

```diff
@@ -25,33 +25,43 @@ int aio_queue_t::submit_batch(aio_iter begin, aio_iter end,
   int r;
 
   aio_iter cur = begin;
-  struct aio_t *piocb[aios_size];
-  int left = 0;
-  while (cur != end) {
-    cur->priv = priv;
-    *(piocb+left) = &(*cur);
-    ++left;
-    ++cur;
-  }
-  ceph_assert(aios_size >= left);
+#if defined(HAVE_LIBAIO)
+  struct aio_t *piocb[max_iodepth];
+#endif
   int done = 0;
-  while (left > 0) {
+  int itemCount0 = 0; //used for LIBAIO only
+  int itemCount = 0;
+  while (cur != end || itemCount0 < itemCount) {
```

Comment:
s/ItemCount/ItemsPulled/
s/ItemCount0/ItemsPushed/
?
But this is obviously just my preference....